### ISSUE TITLE
Between retries, sleep duration is always 0

### ISSUE THREAD

**Issue by yns01** [stripe/stripe-go#651]
It looks like `sleepTime` always return 0. I saw you added `networkRetriesSleep` which is by default set to false. However, since it's not exported and `SetNetworkRetriesSleep` has a do not use comment, we cannot set it to true.

Am i missing something?

```
2018/08/07 10:07:57 Requesting POST api.stripe.com/v1/charges
2018/08/07 10:07:57 Requesting POST api.stripe.com/v1/charges
2018/08/07 10:07:58 Request completed in 1.540859689s (retry: 0)
2018/08/07 10:07:58 Request failed with: {"error":{"code":"idempotency_key_in_use","doc_url":"https://stripe.com/docs/error-codes/idempotency-key-in-use","message":"There is currently another in-progress request using this Idempotent Key (that probably means you submitted twice, and the other request is still going through): 770434815659e5i1m6w. Please try again later.","type":"invalid_request_error"}} (error: <nil>)
2018/08/07 10:07:58 Initiating retry 1 for request POST api.stripe.com/v1/charges after sleeping 0s
2018/08/07 10:07:59 Request completed in 311.409317ms (retry: 1)
2018/08/07 10:07:59 Request failed with: {"error":{"code":"idempotency_key_in_use","doc_url":"https://stripe.com/docs/error-codes/idempotency-key-in-use","message":"There is currently another in-progress request using this Idempotent Key (that probably means you submitted twice, and the other request is still going through): 770434815659e5i1m6w. Please try again later.","type":"invalid_request_error"}} (error: <nil>)
2018/08/07 10:07:59 Initiating retry 2 for request POST api.stripe.com/v1/charges after sleeping 0s
2018/08/07 10:07:59 Request completed in 302.821726ms (retry: 2)
2018/08/07 10:07:59 Request failed with: {"error":{"code":"idempotency_key_in_use","doc_url":"https://stripe.com/docs/error-codes/idempotency-key-in-use","message":"There is currently another in-progress request using this Idempotent Key (that probably means you submitted twice, and the other request is still going through): 770434815659e5i1m6w. Please try again later.","type":"invalid_request_error"}} (error: <nil>)
2018/08/07 10:07:59 Initiating retry 3 for request POST api.stripe.com/v1/charges after sleeping 0s
2018/08/07 10:07:59 Request completed in 233.94586ms (retry: 3)
2018/08/07 10:07:59 Error encountered from Stripe: {"code":"idempotency_key_in_use","status":409,"message":"There is currently another in-progress request using this Idempotent Key (that probably means you submitted twice, and the other request is still going through): 770434815659e5i1m6w. Please try again later.","request_id":"req_LqTR896ZKDVFHS","type":"invalid_request_error"}
Stripe charge failed {"code":"idempotency_key_in_use","status":409,"message":"There is currently another in-progress request using this Idempotent Key (that probably means you submitted twice, and the other request is still going through): 770434815659e5i1m6w. Please try again later.","request_id":"req_LqTR896ZKDVFHS","type":"invalid_request_error"}
```

**Comment by brandur-stripe** [comment]
Well, that's embarrassing.

> Am i missing something?

No, it's just that I forgot to initialize the field to `true` when generating a new `BackendImplementation`. It's Go, so the default value of a boolean is `false`. Added a fix in #653. Thanks for the report.

**Comment by brandur-stripe** [comment]
Fix released in 40.0.1.
